sumpy/recurrence.py

```diff
@@ -0,0 +1,338 @@
+
+    return coeffs
+
+
+def compute_recurrence_relation(coeffs, n_derivs, var):
+    """
+    compute_recurrence_relation
+    Input:
+    - coeffs a 2d array that gives access to the coefficients of poly, where poly
+    represents the coefficients of the ''shifted ODE''
+    (''shifted ODE'' = we substitute all occurences of delta_x with x_0 - c_0)
+    based on their degree as a polynomial in delta_x
+    - n_derivs, the order of the original PDE + 1, i.e. the number of derivatives
+    of f that may be present
+    Output:
+    - a recurrence statement that equals 0 where s(i) is the ith coefficient of
+    the Taylor polynomial for our point potential.
+
+    Description: Takes in coeffs which represents our ``shifted ode in x"
+    (i.e. ode_in_x with coefficients in delta_x) and outputs a recurrence relation
+    for the point potential.
+    """
+    i = sp.symbols("i")
+    s = sp.Function("s")
+    c_vec = make_sympy_vec("c", len(var))
+
+    #Compute symbolic derivative
+    def hc_diff(i, n):
+        retme = 1
+        for j in range(n):
+            retme *= (i-j)
+        return retme
+
+    #We are differentiating deriv_ind, which shifts down deriv_ind.
+    #Do this for one deriv_ind
+    r = 0
+    for deriv_ind in range(n_derivs):
+        part_of_r = 0
+        pow_delta = 0
+        for j in range(len(coeffs[deriv_ind])-1, -1, -1):
+            shift = pow_delta - deriv_ind + 1
+            pow_delta += 1
+            # pylint: disable=not-callable
+            temp = coeffs[deriv_ind][j] * s(i) * hc_diff(i, deriv_ind)
+            part_of_r += temp.subs(i, i-shift)
+        r += part_of_r
+
+    for j in range(1, len(var)):
+        r = r.subs(var[j], c_vec[j])
+
+    return r.simplify()
+
+
+def test_recurrence_finder_laplace():
+    """
+    test_recurrence_finder_laplace
+    Description: Checks that the recurrence finder works correctly for the Laplace
+    2D point potential.
+    """
+    w = make_identity_diff_op(2)
+    laplace2d = laplacian(w)
+    ode_in_r, var, n_derivs = get_pde_in_recurrence_form(laplace2d)
+    ode_in_x = ode_in_r_to_x(ode_in_r, var, n_derivs).simplify()
+    poly = compute_poly_in_deriv(ode_in_x, n_derivs, var)
+    coeffs = compute_coefficients_of_poly(poly, n_derivs)
+    i = sp.symbols("i")
+    s = sp.Function("s")
+    r = compute_recurrence_relation(coeffs, n_derivs, var)
+
+    def coeff_laplace(i):
+        x, y = sp.symbols("x,y")
+        c_vec = make_sympy_vec("c", 2)
+        true_f = sp.log(sp.sqrt(x**2 + y**2))
+        return sp.diff(true_f, x, i).subs(x, c_vec[0]).subs(
+            y, c_vec[1])/math.factorial(i)
+    d = 6
+    # pylint: disable=not-callable
+    val = r.subs(i, d).subs(s(d+1), coeff_laplace(d+1)).subs(
+        s(d), coeff_laplace(d)).subs(s(d-1), coeff_laplace(d-1)).subs(
+            s(d-2), coeff_laplace(d-2)).simplify()
+    assert val == 0
+
+
+def test_recurrence_finder_laplace_three_d():
```

Comment:
Look into `@pytest.mark.parametrize`, generalize `test_recurrence_finder_laplace` to do this, too.

Reply:
I am not sure if it is worth parametrizing these tests since every pde recurrence test requires a different coefficient producing function to check our answers.